build-env/Dockerfile

```diff
@@ -0,0 +1,13 @@
+# build stage
+FROM golang:1.17 as builder
+
+WORKDIR /workspace
+# Copy the Go Module code
+COPY cmd/ ./cmd
+COPY e2e/ ./e2e
+COPY pkg/ ./pkg
+COPY go.mod ./go.mod
```

Comment:
to facilitate container image building caching, shouldn't the correct order be like this?

COPY go.mod ./go.mod
COPY go.sum ./go.sum
RUN go mod download
COPY cmd/ ./cmd
COPY e2e/ ./e2e
COPY pkg/ ./pkg

Reply:
Great callout. Let's incorporate this order into the docker file